### DISCUSSION TITLE
CommunityToolkit.Maui.Storage.FolderPicker to Access to Photos

### DISCUSSION THREAD

**Discussion by ovbarnett** [CommunityToolkit/Maui#1147] [New Feature Discussions]
Hello there;

Is there a way to somehow give CommunityToolkit.Maui.Storage.FolderPicker ability to choose Photos so that the photos/videos taken on iphone/ipad camera would be saved to Photos going forward? Please see attached recording for more details. I got this recording done while looking at the CommunityToolkit.Maui.Storage.FolderPicker sample available in this repository (CommunityToolkit.Maui.Sample.sln).

Thank you for your time and help with this matter!

https://user-images.githubusercontent.com/62110195/233186144-4b0d6ed6-312d-4b4a-953d-13d4fd600855.mp4

**Comment by VladislavAntonyuk** [comment]
On iOS FileSystem is closed. Use MediaPicker

**Reply by ovbarnett** [reply]
@VladislavAntonyuk  before posting this discussion question, I have already looked at the MediaPiker class and it does not seem to satisfy this requirement of our users "photos/videos taken on iphone/ipad camera would be saved to Photos" 
As per this documented example (https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/device-media/picker?view=net-maui-7.0&tabs=android) 

to save the captured photo you need to build the path like so and that's where the problem lies:
// save the file into local storage
            string localFilePath = Path.Combine(FileSystem.CacheDirectory, photo.FileName);
![image](https://user-images.githubusercontent.com/62110195/233427717-d42247fc-d9ca-4f97-99e0-839ac19f428a.png)

FYI, I have already opened up another discussion point with MAUI folks as well re: this exact issue: https://github.com/dotnet/maui/discussions/14674

So, please let's keep this discussion opened and let other folks in the community to chime-in and brainstorm. Hopefully, together we can come up with better and actually helpful idea. Please and thank you.

**Reply by VladislavAntonyuk** [reply]
You can use share API that should allow users save image to Photos
or package like this: https://github.com/dimonovdd/Xamarin.MediaGallery

**Comment by ovbarnett** [comment]
@VladislavAntonyuk I have looked at share API idea already and it does not work as it does not have access to Photos at all.
This suggestion of yours actually works for our users: https://github.com/dimonovdd/Xamarin.MediaGallery
So, let's close this item and thank-you very much for your time and help with this matter!
